src/lib.rs

```diff
@@ -290,16 +290,25 @@ pub trait Handler {
     fn ping_thread(&self, _id: ThreadId) -> Result<(), SimpleError> {
         Err(SimpleError::Error)
     }
+
+    fn read_memory(&self, _address: u64, _length: u64) -> Result<Vec<u8>, SimpleError> {
+        Err(SimpleError::Error)
+    }
+}
+
+fn compute_checksum_incremental(bytes: &[u8], init: u8) -> u8 {
+    bytes.iter().fold(init, |sum, &b| sum.wrapping_add(b))
 }
 
 /// Compute a checksum of `bytes`: modulo-256 sum of each byte in `bytes`.
 fn compute_checksum(bytes: &[u8]) -> u8 {
-    bytes.iter().fold(0, |sum, &b| sum.wrapping_add(b))
+    compute_checksum_incremental(bytes, 0)
 }
 
 enum Response<'a> {
     Empty,
     String(&'a str),
+    Bytes(Vec<u8>),
```

Comment:
Any reason not to use `&'a [u8]` here instead? This enum already has a lifetime, and `String` is using a reference.